src/components/experiment-tracking/run-metadata/index.js

```diff
@@ -20,6 +25,11 @@ const RunMetadata = ({ isSingleRun, runs = [] }) => {
     setToggleNotes({ ...toggleNotes, [index]: !toggleNotes[index] });
   };
 
+  const onTitleOrNoteClick = (id) => {
+    setSelectedMetadataRunId(id);
+    setShowRunDetailsModal(true);
+  };
+
```

Comment:
Similar to my comment above, I would directly assign the run metadata here and remove the need for the `selectedMetadataRunId` hook.

```suggestion
  const onTitleOrNoteClick = (run) => {
    setSelectedRunMetadata(run);
    setShowRunDetailsModal(true);
  };

```

Reply:
Brilliant suggestion. Thanks for this.